cli/ingress-controller/header_roundtripper.go

```diff
@@ -0,0 +1,30 @@
+package main
+
+import (
+	"net/http"
+	"strings"
+)
+
+// HeaderRoundTripper injects Headers into requests
+// made via RT.
+type HeaderRoundTripper struct {
+	headers []string
+	rt      http.RoundTripper
+}
+
+// RoundTrip satisfies the RoundTripper interface.
+func (t *HeaderRoundTripper) RoundTrip(req *http.Request) (*http.Response, error) {
+	newRequest := new(http.Request)
+	*newRequest = *req
+	newRequest.Header = make(http.Header, len(req.Header))
+	for k, s := range req.Header {
+		newRequest.Header[k] = append([]string(nil), s...)
+	}
+	for _, s := range t.headers {
+		split := strings.SplitN(s, ":", 2)
+		if len(split) >= 2 {
```

Comment:
also == 2 here, if we are going to support `:` in value, we could join the slice of split[1:]

Reply:
ditto @fffonion

Reply:
oh didn't notice`SplitN`, that's really convienient